python/cugraph/cugraph/tests/test_node2vec.py

```diff
@@ -172,14 +96,202 @@ def test_node2vec_invalid(
     for bad_depth in invalid_max_depths:
         with pytest.raises(ValueError):
             df, seeds = calc_node2vec(G, start_vertices, max_depth=bad_depth,
-                                      use_padding=use_padding, p=p, q=q)
+                                      compress_result=compress, p=p, q=q)
     # Tests for invalid p
     for bad_p in invalid_pqs:
         with pytest.raises(ValueError):
             df, seeds = calc_node2vec(G, start_vertices, max_depth=max_depth,
-                                      use_padding=use_padding, p=bad_p, q=q)
+                                      compress_result=compress, p=bad_p, q=q)
     # Tests for invalid q
     for bad_q in invalid_pqs:
         with pytest.raises(ValueError):
             df, seeds = calc_node2vec(G, start_vertices, max_depth=max_depth,
-                                      use_padding=use_padding, p=p, q=bad_q)
+                                      compress_result=compress, p=p, q=bad_q)
+
+
+# Remove once bug is resolved
+@pytest.mark.parametrize(*_get_param_args("graph_file", [LINE]))
+@pytest.mark.parametrize(*_get_param_args("directed", DIRECTED_GRAPH_OPTIONS))
+def test_node2vec_line(graph_file, directed):
+    G = utils.generate_cugraph_graph_from_file(graph_file, directed=directed,
+                                               edgevals=True)
+    max_depth = 3
+    start_vertices = [0, 3, 6]
+    df, seeds = calc_node2vec(
+        G,
+        start_vertices,
+        max_depth,
+        compress_result=True,
+        p=0.8,
+        q=0.5
+    )
+
+
+@pytest.mark.parametrize(*_get_param_args("graph_file", utils.DATASETS_SMALL))
+@pytest.mark.parametrize(*_get_param_args("directed", DIRECTED_GRAPH_OPTIONS))
+@pytest.mark.parametrize(*_get_param_args("compress", COMPRESSED))
+def test_node2vec_new(
+    graph_file,
+    directed,
+    compress,
+):
+    cu_M = utils.read_csv_file(graph_file)
+
+    G = cugraph.Graph(directed=directed)
+
+    G.from_cudf_edgelist(cu_M, source="0", destination="1", edge_attr="2",
+                         renumber=False)
+    num_verts = G.number_of_vertices()
+    k = random.randint(6, 12)
+    start_vertices = random.sample(range(num_verts), k)
+    max_depth = 5
+    df, seeds = calc_node2vec(
```

Comment:
Why `df` when it's a tuple of cudf.Series (vertex_paths, edge_weights, vertex_path_sizes)?

Reply:
Yes, that's correct, it's not a dataframe. It's been changed to result